From: SJ Park <sj@kernel.org>
Date: Tue, 25 Aug 2026 17:58:37 -0700
Subject: [PATCH] ==== complement damos quota goal metric ====

mm/damon: introduce damos quota goal target metric complement flag

Aim-oriented DAMOS quota auto-tuning assumes the aggressiveness of the
scheme (quota) and the quota goal metric are directly proportional.
When the scheme's direction is changed, the relationship can be changed.
For example, the quota of a DAMOS scheme that is migrating pages from
node A to node B and node B memory utilization ratio are directly
proportional.  The quota of a DAMOS schme that is migrating pages from
node B to node A and node B memory utilization ratio are not.  Due to
the unflexible assumption, most DAMOS quota tuning goal metrics have a
type for complemented value.  This unnecessarily increase the number of
metrics to support.

Add a new flag, complement, to allow flexible tuning goal setup.  The
flag decide whether to use the complemented value of the given goal
metric for the tuning.  For example, if the complement flag is set, node
B memory utilization ratio metric works same to node B free memory ratio
metric.

Signed-off-by: SJ Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_e24u9vs6 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_e24u9vs6

diff --git a/hkml_cv_bogus/hkml_cv_bogus_e24u9vs6 b/hkml_cv_bogus/hkml_cv_bogus_e24u9vs6
new file mode 100644
index 0000000000000..e69de29bb2d1d
-- 
2.47.3

